packages/embark/src/lib/utils/template_generator.js

```diff
@@ -73,52 +74,150 @@ class TemplateGenerator {
     }
-    utils.extractZip(tmpFilePath, fspath, {
-      map: file => {
-        let fixed_path = file.path.split('/');
-        fixed_path.shift(); // remove first directory
-        file.path = utils.joinPath(...fixed_path);
-        return file;
-      }
-    }, () => {
-      self.installTemplate(fspath, name, true);
+    this.extract(tmpFilePath, fspath, () => {
+      this.installTemplate(fspath, name, true);
     });
   }
 
+  extract(filePath, destinationFolder, cb = () => {}) {
+    utils.extractZip(
+      filePath,
+      destinationFolder,
+      {
+        map: file => {
+          let fixed_path = file.path.split('/');
+          fixed_path.shift(); // remove first directory
+          file.path = utils.joinPath(...fixed_path);
+          return file;
+        }
+      },
+      cb
+    );
+  }
+
   generate(destinationFolder, name) {
     const fspath = utils.joinPath(destinationFolder, name);
     this.checkPathExists(fspath);
+
     console.log(__('Initializing Embark template...').green);
-    let templatePath = fs.embarkPath(utils.joinPath('templates', this.templateName));
-    fs.copySync(templatePath, fspath);
-
-    this.installTemplate(
-      fspath,
-      name,
-      (this.templateName === 'boilerplate' || this.templateName === 'demo'),
-      () => {
-        if (name === 'embark_demo') {
-          console.log('-------------------'.yellow);
-          console.log(__('Next steps:').green);
-          console.log(('-> ' + ('cd ' + fspath).bold.cyan).green);
-          console.log('-> '.green + 'embark run'.bold.cyan);
-          console.log(__('For more info go to http://embark.status.im').green);
+
+    const templatePkg = `embark-dapp-template-${this.templateName}`;
+    let templateSpecifier;
+    const monorepoRoot = this.insideMonorepo();
+    if (monorepoRoot) {
+      templateSpecifier = utils.joinPath(
+        monorepoRoot, 'dapps/templates', this.templateName
+      );
+    } else {
+      const version = fs.readJSONSync(fs.embarkPath('package.json')).version;
+      templateSpecifier = `${templatePkg}@^${version}`;
+    }
+
+    const tmpDir = require('fs-extra').mkdtempSync(
+      utils.joinPath(require('os').tmpdir(), `${this.templateName}-`)
+    );
+
+    execSync(`npm pack ${templateSpecifier}`, {cwd: tmpDir, stdio: 'ignore'});
+    const packed = utils.filesMatchingPattern(
+      [utils.joinPath(tmpDir, '*.tgz')]
+    )[0];
+
+    this.extract(packed, fspath, () => {
+      this.installTemplate(
+        fspath,
+        name,
+        true,
+        () => {
+          if (name === 'embark_demo') {
+            console.log('-------------------'.yellow);
+            console.log(__('Next steps:').green);
+            console.log(('-> ' + ('cd ' + fspath).bold.cyan).green);
+            console.log('-> '.green + 'embark run'.bold.cyan);
+            console.log(__('For more info go to http://embark.status.im').green);
+          }
+        }
+      );
+    });
+  }
+
+  insideMonorepo() {
```

Comment:
Just a suggestion, I like to see methods that return bools prefixed with a word that allows us to assume it's type, ie "is", "has", can", etc. Additionally, for these types of class-specific functions, it makes it easy to read when they are properties of the class. What I'm suggesting would ultimately look like:

```
get isInsideMonorepo() {
  // ...
}
```
which would be called using:
```
if(this.isInsideMonorepo) {
  // ...
}
```
Complex functionality can additionally be supported by backing variables, preferrably prefixed with an underscore, ie `this._isInsideMonorepo` and the value can be checked before executing the method, ie:
```
class TemplateGenerator {
  constructor() {
    this._isInsideMonorepo = null;
  }

  get isInsideMonorepo() {
    if (this._isInsideMonorepo !== null) {
      return this._isInsideMonorepo;
    }
    // ... do work to determine if inside monorepo
  }
}
```

Reply:
Sorry, now I realise this is meant to return a string. 

So it would look like 
```
class TemplateGenerator {
  constructor() {
    this._monorepoRootPath = "";
  }

  get monorepoRootPath() {
    if (this._monorepoRootPath) {
      return this._monorepoRootPath;
    }
    // ... do work to determine monorepo root path
  }
}
```
`insideMonorepo` doesn't scream "I'm returning a monorepo root path".

Reply:
Good point, I'll make the change.

Reply:
Done.